perf_workflow/run_perf_suite/ccr_perf_test.py

```diff
@@ -0,0 +1,294 @@
+
+    def verify_result(self, result, leaderCluster, followerCluster):
+        self.verify_checkpoints(followerCluster)
+        self.verify_doc_count(leaderCluster, followerCluster)
+        self.execution_id = result["testExecutionId"]
+        for telemetry_result in result["testResults"]["customTelemetryData"]["overall"]:
+            if (
+                telemetry_result["telemetryDevice"] == "ccr-stats"
+                and telemetry_result["metric"] == "replication_lag"
+            ):
+                # TODO: Add support for comparison with benchmark run
+                assert (
+                    telemetry_result["p100"] > 0
+                ), "Replication lag should be non-zero"
+                assert (
+                    telemetry_result["p100"] < self.MAX_ALLOWED_REPLICATION_LAG_SECONDS
+                ), "Replication lag p100 was greater than a minute"
+                self.test_succeeded = True
+                logging.info(f"Test succeeded for {self.description}")
+
+    def verify_doc_count(self, leaderCluster, followerCluster):
+        index_name = self.test_config["Workload"]
+        path = f"/{index_name}/_count?format=json"
+
+        leader_url = "".join([leaderCluster.endpoint_with_port(), path])
+        leader_resp = requests.get(url=leader_url, auth=self.CREDS, verify=False)
+        leader_doc_count = leader_resp.json()['count']
+
+        follower_url = "".join([followerCluster.endpoint_with_port(), path])
+        follower_resp = requests.get(url=follower_url, auth=self.CREDS, verify=False)
+        follower_doc_count = follower_resp.json()['count']
+
+        assert leader_doc_count == follower_doc_count, "Doc count on leader doesnt match with follower"
+
+    def verify_checkpoints(self, followerCluster):
+        index_name = self.test_config["Workload"]
+        path = f"/_plugins/_replication/{index_name}/_status?pretty"
+
+        url = "".join([followerCluster.endpoint_with_port(), path])
+        follower_resp = requests.get(url=url, auth=self.CREDS, verify=False)
+        leader_checkpoint = follower_resp.json()['syncing_details']['leader_checkpoint']
+        follower_checkpoint = follower_resp.json()['syncing_details']['follower_checkpoint']
+
+        assert leader_checkpoint == follower_checkpoint, "Follower is not at same checkpoint as leader"
+
+    def setup_logging(self):
+        root = logging.getLogger()
+        root.setLevel(logging.INFO)
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setLevel(logging.INFO)
+        root.addHandler(handler)
+
+    def workload_options(self, leaderCluster, followerCluster):
+        telemetry_options = {
+            "ccr-stats-sample-interval": 1,
+            "ccr-stats-indices": {"follower": [self.test_config["Workload"]]},
+        }
+        workload_options = {
+            "telemetry-params": json.dumps(telemetry_options),
+            "workload-params": "number_of_shards:{},number_of_replicas:{}".format(
```

Comment:
Could we pass codec index setting as well to force the default (optimized for speed)?

Reply:
Tried adding this in workload params but the OSB test run failed. Will try to get that fixed and do this change in a follow up PR.